R/scale-expansion.r

```diff
@@ -200,21 +200,28 @@ expand_limits_continuous_trans <- function(limits, expand = expansion(0, 0),
 expand_limits_discrete_trans <- function(limits, expand = expansion(0, 0),
                                          coord_limits = c(NA, NA), trans = identity_trans(),
                                          range_continuous = NULL) {
+  if (is.discrete(limits) || length(limits) == 0) {
+    n_discrete_limits <- length(limits)
+  } else {
+    warn(
+      "Continuous limits supplied to discrete scale.\nDid you mean `limits = factor(...)` or `scale_*_continuous()`?"
```

Comment:
I wonder whether this warning is in the wrong place. It's making statements about scale functions and their arguments. Shouldn't it be located inside the appropriate scale functions? (I.e., check the `limit` argument to `scale_*_discrete()` rather than checking here.)

Reply:
Yes, `discrete_scale()` seems the right place.